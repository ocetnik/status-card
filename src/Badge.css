.Badge-wrapper {
  padding: 21px 12px;
}

.Badge {
  border-radius: 50%;
  height: 8px;
  width: 8px;
}

.Badge-inside-image {
  border: 2px solid white;
  position: absolute;
  right: -2px;
  top: 20px;
}
