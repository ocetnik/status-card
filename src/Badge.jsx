import React from 'react';

import './Badge.css';

const WithWrapper = ({ renderImage, children }) =>
  renderImage ? children : <div className="Badge-wrapper">{children}</div>;

export default ({ renderImage, circleColor }) => (
  <WithWrapper renderImage={renderImage}>
    <div
      className={`Badge${renderImage ? ' Badge-inside-image' : ''}`}
      style={{ backgroundColor: circleColor }}
    />
  </WithWrapper>
);
