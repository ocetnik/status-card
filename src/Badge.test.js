import React from 'react';
import renderer from 'react-test-renderer';

import Badge from './Badge';

it('renders with wrapper', () => {
  const tree = renderer
    .create(<Badge children={<div>badge</div>} renderImage={false} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders without wrapper', () => {
  const tree = renderer
    .create(<Badge children={<div>badge</div>} renderImage={true} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
