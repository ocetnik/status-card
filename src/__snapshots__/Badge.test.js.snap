// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with wrapper 1`] = `
<div
  className="Badge-wrapper"
>
  <div
    className="Badge"
    style={
      Object {
        "backgroundColor": undefined,
      }
    }
  />
</div>
`;

exports[`renders without wrapper 1`] = `
<div
  className="Badge Badge-inside-image"
  style={
    Object {
      "backgroundColor": undefined,
    }
  }
/>
`;
